### PR TITLE
feat: tezos-provider

### PR DESCRIPTION
**Pull request by dianasavvatina**
## Description

Adding a TezosProvider using EthereumProvider as a reference.
See README for details.

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?

To test:
```
cd providers/tezos-provider
npm test
```

## Checklist

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: providers/tezos-provider/src/TezosProvider.ts [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+        entrypoint,
+        value: { prim: "Unit" },
+      },
+    };
+  }
+
+  public async tezosSendStake(op: PartialTezosTransactionOperation = {
+    kind: TezosOperationType.TRANSACTION,
+    destination: this.address ?? "[initiate provider first]",
+    amount: "1",
```
Comment (ajinkyasraj) [inline providers/tezos-provider/src/TezosProvider.ts]: Amount is not an optional field. The user must provide this while creating the operation.

File: providers/tezos-provider/src/TezosProvider.ts [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+      value: { prim: "Unit" },
+    },
+  }): Promise<TezosSendResponse> {
+    return this.tezosSend(this.createStakingOperation("stake", op));
+  }
+
+  public async tezosSendUnstake(op: PartialTezosTransactionOperation = {
+    kind: TezosOperationType.TRANSACTION,
+    destination: this.address ?? "[initiate provider first]",
+    amount: "1",
```
Comment (ajinkyasraj) [inline providers/tezos-provider/src/TezosProvider.ts]: Same here.

File: providers/tezos-provider/src/TezosProvider.ts (hunk 1 above)
Comment (ajinkyasraj) [reply providers/tezos-provider/src/TezosProvider.ts]: Do not provide default value for amount. Let the transaction fail if the amount is not set in `op`

File: providers/tezos-provider/src/TezosProvider.ts [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+  TezosActivateAccountOperation, 
+  TezosAttestationOperation, 
+  TezosAttestationWithSlotOperation, 
+  TezosBallotOperation, 
+  TezosDoubleAttestationEvidenceOperation, 
+  TezosDoubleBakingEvidenceOperation, 
+  TezosDoublePreAttestationEvidenceOperation, 
+  TezosDoublePreEndorsementEvidenceOperation, 
+  TezosDrainDelegateOperation, 
+  TezosEndorsementOperation,
```
Comment (ajinkyasraj) [inline providers/tezos-provider/src/TezosProvider.ts]: (pre)Attestation and Endorsement can not be done via wallet/dapp. These are baker specific operation and they are sent via the actual octez binary. Can be removed from the list. Also DrainDelegateOperation is to be removed. Thats a security sensetive operation, need not be provided on wallets. Same goes for DoubleBaking/Attestation evidence operations. They are pushed by the octez-accuser binary.
SeedNonceRevelation and VdfRevelation are also baker specific. Need not be included in wallets.

Comment (dianasavvatina) [reply providers/tezos-provider/src/TezosProvider.ts]: fixed

File: providers/tezos-provider/src/TezosProvider.ts (hunk 2 above)
Comment (dianasavvatina) [reply providers/tezos-provider/src/TezosProvider.ts]: fixed

File: providers/tezos-provider/src/TezosProvider.ts (hunk 1 above)
Comment (dianasavvatina) [reply providers/tezos-provider/src/TezosProvider.ts]: fixed

File: providers/tezos-provider/src/TezosProvider.ts (hunk 3 above)
Comment (dianasavvatina) [reply providers/tezos-provider/src/TezosProvider.ts]: Introduced `UnsupportedOperations`

File: providers/tezos-provider/test/samples.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  kind: TezosOperationType.TRANSACTION,
+  destination:"[own adress]",
+  amount: "0",
+  parameters: {
+    entrypoint: "finalize_unstake",
+    value: { prim: "Unit" },
+  },
+};
+
+const TezosIncreasePaidStorageOperation: PartialTezosIncreasePaidStorageOperation = {
```
Comment (ajinkyasraj) [inline providers/tezos-provider/test/samples.ts]:
```suggestion
const tezosIncreasePaidStorageOperation: PartialTezosIncreasePaidStorageOperation = {
```

**Review by ajinkyasraj — APPROVED** [review]
LGTM

**Pull request merged**